src/libstd/sys/windows/rand.rs

```diff
@@ -1,14 +1,16 @@
 use crate::io;
 use crate::mem;
 use crate::sys::c;
+use crate::ptr;
 
 pub fn hashmap_random_keys() -> (u64, u64) {
     let mut v = (0, 0);
     let ret = unsafe {
-        c::RtlGenRandom(&mut v as *mut _ as *mut u8,
```

Comment:
Ah sorry for leading you astray here, but IIRC the usage of `RtlGenRandom` is pretty specific to work in as many cases as possible (this came up in the context of Firefox awhile back I believe), so I think we'll want to stick to `RtlGenRandom` for normal Windows platforms but use `BCryptGenRandom` on UWP

Reply:
Will do!

Reply:
It's now back to a UWP version and a "regular" version

Reply:
@alexcrichton
Should we keep `RtlGenRandom` in `getrandom` as well (see rust-random/getrandom#65)? IIUC Rust and Firefox do not support Windows XP anymore and `BCryptGenRandom` should be available on Windows Vista and later.

Reply:
Since this is a new platform inherited knowledge from other platforms I don't think applies. If RtlGenRandom doesn't exist on UWP then there's not really a question, I think something else basically has to be used.

Reply:
I mean that `RtlGenRandom` looks like a legacy interface. Citing this [message](https://github.com/rust-random/getrandom/issues/65#issuecomment-515436418):
>Microsoft advises to use `CryptGenRandom` over `RtlGenRandom`, but `CryptGenRandom` is deprecated and the "next generation cryptography API" should be used instead, and `BCryptGenRandom` is part of that API

So the question is not about UWP, but about non-UWP. If I am not mistaken `BCryptGenRandom` should be present for all Windows versions supported by Rust and Firefox, so shouldn't we switch to it?

Reply:
Ah ok, it's probably best then to have this discussion not on a PR that's only about UWP? 

I'd recommend reading up on the history of how we settled on this function, IIRC it's an oddly storied history.

Reply:
Yes, I agree, this is why I've linked the `getrandom` issue. :) But I am not sure where i should start the reading, can you please link some starting points in the issue?